src/app/navigation/navigation.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component, inject } from '@angular/core';
+import { BreakpointObserver, Breakpoints } from '@angular/cdk/layout';
+import { Observable } from 'rxjs';
+import { map, shareReplay } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-navigation',
+  templateUrl: './navigation.component.html',
+  styleUrls: ['./navigation.component.css']
+})
+export class NavigationComponent {
+  private breakpointObserver = inject(BreakpointObserver);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `inject` function is not the correct way to inject dependencies in Angular. You should use the constructor for dependency injection. Here's how you can do it:

```diff
- private breakpointObserver = inject(BreakpointObserver);
+ constructor(private breakpointObserver: BreakpointObserver) {}
```
This change will ensure that the `BreakpointObserver` service is properly injected into your component.


<!-- This is an auto-generated comment by OSS CodeRabbit -->